src/borg/hashindex.pyx

```diff
@@ -92,7 +92,6 @@ cdef class IndexBase:
                     self.index = hashindex_read(fd, permit_compact)
             else:
                 self.index = hashindex_read(path, permit_compact)
-            assert self.index, 'hashindex_read() returned NULL with no exception set'
```

Comment:
I was going to say that probably Cython generates code to handle this case, but in fact the /*except:*/ code relies on an exception being set and the branch to it only depends on ret==NULL. Btw., this patch produces virtually the same code as before, including the temporary, so it won't fix the false positive of that tool.